addons/controllers/addon_controller.go

```diff
@@ -311,6 +314,9 @@ func (r *AddonReconciler) reconcileNormal(
 		return ctrl.Result{}, kerrors.NewAggregate(errors)
 	}
 
+	if !controllerutil.ContainsFinalizer(cluster, addontypes.AddonFinalizer) {
+		controllerutil.AddFinalizer(cluster, addontypes.AddonFinalizer)
```

Comment:
Does this call update/patch the object. I dont think it does that.

Reply:
fixed